rplugin/python3/LanguageClient/DiagnosticsDisplay.py

```diff
@@ -1,23 +1,23 @@
 DiagnosticsDisplay = {
-    1: {
+    "1": {
```

Comment:
Thanks for the work.

Can you explain why this change is necessary?

I am asking because this change would break backward compatibility. User can [customize this display](https://github.com/autozimu/LanguageClient-neovim/blob/master/doc/LanguageClient.txt#L70).

Reply:
Ah! I made that change because the integration tests read in the state (including the diagnostics) as a [json payload](https://github.com/autozimu/LanguageClient-neovim/blob/master/rplugin/python3/LanguageClient/LanguageClient_test.py#L47), which only supports string keys, and merges it with the default state containing numbered keys. So the merged end-result DiagnosticsDisplay was 
`{ 1: ..., 2: ..., 3: ..., 4: ..., "1": ..., "2": ..., "3": ..., "4": ... }`
which seemed unintended and threw off the total diagnostic count I was using to calculate sign ids.

I'll simply make a list of the `name`s of a DiagnosticDisplay (excluding duplicates) and use that to calculate an offset instead.